app/jobs/ingest_archival_media_bag_job.rb

```diff
@@ -0,0 +1,227 @@
+      end
+
+      private
+
+        # create an AudioPath object for each audio file
+        def audio_files
+          @audio_files ||= path.join("data").each_child.select { |file| [".wav", ".mp3"].include? file.extname }.map { |file| IngestableAudioFile.new(path: file) }
+        end
+    end
+
+    # Decorates the Pathname with some convenience parsing methods
+    # Provides methods needed by FileMetadata.for
+    class IngestableAudioFile
+      attr_reader :path, :barcode_with_part
+      def initialize(path:)
+        @path = path
+      end
+
+      def original_filename
+        path.split.last
+      end
+
+      def mime_type
+        if master? || intermediate?
+          "audio/wav"
+        else
+          "audio/mpeg"
+        end
+      end
+      alias content_type mime_type
+
+      def use
+        if master?
+          Valkyrie::Vocab::PCDMUse.PreservationMasterFile
+        elsif intermediate?
+          Valkyrie::Vocab::PCDMUse.IntermediateFile
+        elsif access?
+          Valkyrie::Vocab::PCDMUse.ServiceFile
+        end
+      end
+
+      def master?
+        path.fnmatch "*_pm.wav"
+      end
+
+      def intermediate?
+        path.fnmatch "*_i.wav"
+      end
+
+      def access?
+        path.fnmatch "*_a.mp3"
+      end
+
+      def barcode_with_part
+        @barcode_with_part ||= BARCODE_WITH_PART_REGEX.match(original_filename.to_s)[1]
+      end
+
+      def barcode
+        barcode_with_part.split('_').first
+      end
+
+      def part
+        barcode_with_part.split('_').last
+      end
+    end
+
+    # Wraps a hash of component_id => barcode_with_part
+    class BarcodeComponentDict
+      attr_reader :collection, :dict
+      def initialize(collection)
+        @collection = collection
+        parse_dict
+      end
+
+      def lookup(barcode)
+        @dict[barcode]
+      end
+
+      private
+
+        # query the EAD for filenames, navigates back up to their component IDs
+        def parse_dict
+          @dict = {}
+          barcode_nodes.each do |node|
+            @dict[get_barcode(node)] = get_id(node)
+          end
+        end
+
+        def xml
+          @xml ||= Nokogiri::XML(collection.imported_metadata.first.source_metadata.first).remove_namespaces!
+        end
+
+        def barcode_nodes
+          xml.xpath('//altformavail/p')
+        end
+
+        def get_id(node)
+          node.parent.parent.attributes['id'].value
+        end
+
+        def get_barcode(node)
+          BARCODE_WITH_PART_REGEX.match(node.content)[1]
+        end
+    end
+
+    class Ingester
+      delegate :query_service, to: :adapter
+      delegate :persister, to: :adapter
+
+      attr_reader :collection, :bag, :user, :adapter
+      def initialize(collection:, bag:, user:, adapter:)
+        @collection = collection
+        @bag = bag
+        @user = user
+        @adapter = adapter
+      end
+
+      def ingest
+        component_dict = BarcodeComponentDict.new(collection)
```

Comment:
Extract to instance method

Reply:
@tpendragon just like `component_dict = get_component_dict(collection)`?

Reply:
Sorry - something like
```
def component_dict
  @component_dict ||= get_component_dict(collection)
end
```